src/main/java/com/github/onsdigital/thetrain/App.java

```diff
@@ -50,107 +41,127 @@
 
 public class App {
 
-    static Map<String, String> ROUTES;
-
+    /**
+     * Start The Train.
+     * A {@link LoggingException} throw when attempting to init the application is considered fatal and will result
+     * in the app exiting with a system exit code of 1 - not being able to log means we are helpless if a publish
+     * were to fail.
+     *
+     * @param args
+     */
     public static void main(String[] args) {
-        LogSerialiser serialiser = new JacksonLogSerialiser();
-        LogStore store = new MDCLogStore(serialiser);
-        Logger logger = new LoggerImpl("the-train");
-
         try {
-            DPLogger.init(new Builder()
-                    .logger(logger)
-                    .logStore(store)
-                    .serialiser(serialiser)
-                    .dataNamespace("train.data")
-                    .create());
+            startApp();
         } catch (LoggingException ex) {
             System.err.println(ex);
             System.exit(1);
-        }
-
-        try {
-            info().log("starting the-train");
-            allAboard();
         } catch (Exception e) {
             fatal(e).log("unexpected error while attempting to start the-train");
             System.exit(1);
         }
     }
 
-    private static void allAboard() throws Exception {
-        AppConfiguration config = AppConfiguration.get();
 
-        // init services.
-        Publisher.init(config.publishThreadPoolSize());
-        Transactions.init(config.transactionStore());
+    private static void startApp() throws Exception {
+        initLogging();
 
-        ROUTES = new HashMap<>();
+        info().log("starting the-train");
 
+        AppConfiguration config = AppConfiguration.get();
+        initServices(config);
         port(config.port());
+        registerHTTPFilters();
+        registerExeptionHandlers();
+        registerEndpoints(config);
 
-        before("/*", new BeforeFilter());
+        info().data("PORT", config.port()).log("train start up completed successfully");
+    }
 
-        AfterFilter afterFilter = new AfterFilter();
-        after("/*", afterFilter);
+    private static void initLogging() throws LoggingException {
+        LogSerialiser serialiser = new JacksonLogSerialiser();
+        LogStore store = new MDCLogStore(serialiser);
+        Logger logger = new LoggerImpl("the-train");
 
-        registerExeptionHandlers();
+        DPLogger.init(new Builder()
```

Comment:
The only checked error thrown by this code is a `LoggingException`.  Unlike Go You don't have to return errors - you either `catch` and handle them or declare the method throws it
```
private static void initLogging() throws LoggingException {
...
}
```
This is telling the caller that if  this error occurs it won't handle it and throw it back to the caller making it its responsibly to handle - either catch it and deal with it or throw it again (this can happen multiple times until something on the call stack will handle it).

If anything goes wrong here the exception is thrown and not handled and is propagated back up to `main`. The `try-catch` block in main is explicitly catching  errors of type LoggingException. If this happens it catches it, logs the details and does a os exit.